### PR TITLE
Teach //flutter/lib/ui to build on Fuchsia

### PR DESCRIPTION
**Pull request by abarth**
This patch removes the //flutter/sky/engine dependency from //flutter/lib/ui,
which lets us build the bulk of the dart:ui library without needing to build
//flutter/sky/engine.

### CODE REVIEW

**Comment by abarth** [comment]
@chinmaygarde

File: tonic/dart_state.h [hunk 1]
```diff
@@ -21,7 +24,17 @@ class DartState : public tonic::DartState {
   static DartState* From(Dart_Isolate isolate);
   static DartState* Current();
 
+  const ftl::RefPtr<ftl::TaskRunner>& ui_task_runner() {
```
Comment (chinmaygarde) [inline tonic/dart_state.h]: Having the UI and IO task runners hosted in the "DartState" seems a but awkward.

**Comment by chinmaygarde** [comment]
lgtm

File: tonic/dart_state.h (hunk 1 above)
Comment (abarth) [inline tonic/dart_state.h]: Yeah, I thought about moving where the static is defined so I could see it from image_decoders.cc.

**Pull request merged**